Compilation error with YARP 4

## Task description

Despite:
- https://github.com/mesh-iit/ergocub-software/pull/340

ergocub-software is not compiling under YARP 4:

```
> cmake --build $BUILD_SUFFIX -- -j$(nproc)
[ 12%] Building CXX object src/tools/joint-pid-tuning/cpp/joint-position-move/CMakeFiles/joint-position-move.dir/main.cpp.o
[ 25%] Building CXX object src/tools/joint-pid-tuning/cpp/joint-open-loop-move/CMakeFiles/joint-open-loop-move.dir/main.cpp.o
[ 37%] Building CXX object src/modules/couplingXCubHandMk5/CMakeFiles/yarp_couplingXCubHandMk5.dir/yarp_plugin_couplingXCubHandMk5.cpp.o
[ 62%] Building CXX object src/modules/couplingXCubHandMk5/CMakeFiles/yarp_couplingXCubHandMk5.dir/CouplingXCubHandMk5_ParamsParser.cpp.o
[ 62%] Building CXX object src/modules/couplingXCubHandMk5/CMakeFiles/yarp_couplingXCubHandMk5.dir/CouplingXCubHandMk5.cpp.o
/mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp: In function ‘bool resetPosition(uint8_t)’:
/mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp:129:30: error: cannot bind non-const lvalue reference of type ‘bool&’ to a value of type ‘bool*’
  129 |     iPos->checkMotionDone(j, &done);
      |                              ^~~~~
In file included from /mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp:19:
/mnt/data/src/GIT/robotology/yarp/src/libYARP_dev/src/yarp/dev/IPositionControl.h:303:65: note:   initializing argument 2 of ‘virtual yarp::dev::ReturnValue yarp::dev::IPositionControl::checkMotionDone(int, bool&)’
  303 |     virtual yarp::dev::ReturnValue checkMotionDone(int j, bool& flag)=0;
      |                                                           ~~~~~~^~~~
/mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp: In function ‘int main(int, char**)’:
/mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp:100:41: error: cannot bind non-const lvalue reference of type ‘bool&’ to a value of type ‘bool*’
  100 |         iPos->checkMotionDone(joint_id, &done);
      |                                         ^~~~~
In file included from /mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp:20:
/mnt/data/src/GIT/robotology/yarp/src/libYARP_dev/src/yarp/dev/IPositionControl.h:303:65: note:   initializing argument 2 of ‘virtual yarp::dev::ReturnValue yarp::dev::IPositionControl::checkMotionDone(int, bool&)’
  303 |     virtual yarp::dev::ReturnValue checkMotionDone(int j, bool& flag)=0;
      |                                                           ~~~~~~^~~~
/mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp:137:53: error: cannot bind non-const lvalue reference of type ‘bool&’ to a value of type ‘bool*’
  137 |                     iPos->checkMotionDone(joint_id, &done);
      |                                                     ^~~~~
/mnt/data/src/GIT/robotology/yarp/src/libYARP_dev/src/yarp/dev/IPositionControl.h:303:65: note:   initializing argument 2 of ‘virtual yarp::dev::ReturnValue yarp::dev::IPositionControl::checkMotionDone(int, bool&)’
  303 |     virtual yarp::dev::ReturnValue checkMotionDone(int j, bool& flag)=0;
      |                                                           ~~~~~~^~~~
/mnt/data/src/GIT/mesh-iit/ergocub-software/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp:181:41: error: cannot bind non-const lvalue reference of type ‘bool&’ to a value of type ‘bool*’
  181 |         iPos->checkMotionDone(joint_id, &done);
      |                                         ^~~~~
/mnt/data/src/GIT/robotology/yarp/src/libYARP_dev/src/yarp/dev/IPositionControl.h:303:65: note:   initializing argument 2 of ‘virtual yarp::dev::ReturnValue yarp::dev::IPositionControl::checkMotionDone(int, bool&)’
  303 |     virtual yarp::dev::ReturnValue checkMotionDone(int j, bool& flag)=0;
      |                                                           ~~~~~~^~~~
gmake[2]: *** [src/tools/joint-pid-tuning/cpp/joint-position-move/CMakeFiles/joint-position-move.dir/build.make:76: src/tools/joint-pid-tuning/cpp/joint-position-move/CMakeFiles/joint-position-move.dir/main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:339: src/tools/joint-pid-tuning/cpp/joint-position-move/CMakeFiles/joint-position-move.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake[2]: *** [src/tools/joint-pid-tuning/cpp/joint-open-loop-move/CMakeFiles/joint-open-loop-move.dir/build.make:76: src/tools/joint-pid-tuning/cpp/joint-open-loop-move/CMakeFiles/joint-open-loop-move.dir/main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:313: src/tools/joint-pid-tuning/cpp/joint-open-loop-move/CMakeFiles/joint-open-loop-move.dir/all] Error 2
[ 75%] Linking CXX shared module ../../../lib/yarp/yarp_couplingXCubHandMk5.so
[ 75%] Built target yarp_couplingXCubHandMk5
gmake: *** [Makefile:136: all] Error 2
 
```

## Definition of Done

It compiles fine with YARP 4

cc @elandini84

--- a/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp
+++ b/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp
@@ -126,7 +126,11 @@ bool resetPosition(uint8_t j) {
   bool done = false;
 
   while (!done) {
+    #if YARP_VERSION_MAJOR >= 4
     iPos->checkMotionDone(j, done);
+    #else
+    iPos->checkMotionDone(j, &done);
+    #endif
     Time::yield();
   }
 

--- a/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp
+++ b/src/tools/joint-pid-tuning/cpp/joint-open-loop-move/main.cpp
@@ -52,7 +52,7 @@ private:
   bool done_l = false;
   DataExperiment data;
   uint8_t j = 0;
-  std::atomic<double> pwm_{0.0};  
+  std::atomic<double> pwm_{0.0};
   double thr_min = 0;
   double thr_max = 0;
   double t0 = 0;
@@ -126,7 +126,7 @@ bool resetPosition(uint8_t j) {
   bool done = false;
 
   while (!done) {
-    iPos->checkMotionDone(j, &done);
+    iPos->checkMotionDone(j, done);
     Time::yield();
   }
 

--- a/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp
+++ b/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp
@@ -97,7 +97,7 @@ int main(int argc, char * argv[])
     auto cycle_done{ false };
 
     while (!done) {
-        iPos->checkMotionDone(joint_id, &done);
+        iPos->checkMotionDone(joint_id, done);
         Time::yield();
     }
 
@@ -134,7 +134,7 @@ int main(int argc, char * argv[])
                 iEnc->getEncoder(joint_id, &data.enc);
 
                 if (Time::now() - t1 >= 0.01) {
-                    iPos->checkMotionDone(joint_id, &done);
+                    iPos->checkMotionDone(joint_id, done);
                     t1 = Time::now();
 
                     if(done)
@@ -178,7 +178,7 @@ int main(int argc, char * argv[])
     #endif
     iPos->positionMove(joint_id, 0.);
     while (!done) {
-        iPos->checkMotionDone(joint_id, &done);
+        iPos->checkMotionDone(joint_id, done);
         Time::yield();
     }
 

--- a/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp
+++ b/src/tools/joint-pid-tuning/cpp/joint-position-move/main.cpp
@@ -97,7 +97,11 @@ int main(int argc, char * argv[])
     auto cycle_done{ false };
 
     while (!done) {
+        #if YARP_VERSION_MAJOR >= 4
         iPos->checkMotionDone(joint_id, done);
+        #else
+        iPos->checkMotionDone(joint_id, &done);
+        #endif
         Time::yield();
     }
 
@@ -134,7 +138,11 @@ int main(int argc, char * argv[])
                 iEnc->getEncoder(joint_id, &data.enc);
 
                 if (Time::now() - t1 >= 0.01) {
+                    #if YARP_VERSION_MAJOR >= 4
                     iPos->checkMotionDone(joint_id, done);
+                    #else
+                    iPos->checkMotionDone(joint_id, &done);
+                    #endif
                     t1 = Time::now();
 
                     if(done)
@@ -178,7 +186,11 @@ int main(int argc, char * argv[])
     #endif
     iPos->positionMove(joint_id, 0.);
     while (!done) {
+        #if YARP_VERSION_MAJOR >= 4
         iPos->checkMotionDone(joint_id, done);
+        #else
+        iPos->checkMotionDone(joint_id, &done);
+        #endif
         Time::yield();
     }
 

